src/pybind/mgr/rook/rook_cluster.py

```diff
@@ -245,7 +261,7 @@ def add_nfsgw(self, spec):
                 "namespace": self.rook_namespace
             },
             "spec": {
-                "RADOS": {
```

Comment:
is there any way we can validate the CRD without a cluster? e.g. in a unit test?

Reply:
@travisn ?

Reply:
I'd suggest a unit test that at least checks if the json is valid. More validation that that would require spinning up rook. This reminds me we should add crd validation in the crd definition as well.

Reply:
The only other alternative I could think of would be a `rook-ceph-client` python module that is part of the rook project and tested by rook, instead of the ceph project.

Reply:
The issue here is that the json is embedded into the code. I'm not sure how we'd unit test that at all.

I suppose we could break out the code that sets up the dict into a separate function, call that and then validate the result somehow. I've no idea how we'd keep such a thing in sync with rook though.

Reply:
honestly, I thing this code belongs into the rook project git.